app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,84 @@
+class TweetsController < ApplicationController
+  
+  get '/users/:slug' do
+    @user = User.find_by_slug(params[:slug])
+    if  @user
+      erb :'tweets/show_tweet'
+    end
+  end
+
+  get '/tweets/new' do
+    if logged_in?
+      @user = User.find(session[:id])
+      erb :'tweets/create_tweet'
+    else
+      redirect '/login'
+    end
+  end
+
+
+  get '/tweets' do
+    if !logged_in?
+      redirect '/login'
+    else
+      @user = User.find(session[:id])
+      erb :home
+    end
+  end
+
+  get '/tweets/:id' do
+    if logged_in?
+      @tweet = Tweet.find(params[:id])
+      erb :'tweets/show_tweet'
+    else
+      redirect '/login'
+    end
+  end
+
+
+  get '/tweets/:id/edit' do
+    if logged_in?
+      @tweet = Tweet.find(params[:id])
+      erb :'tweets/edit_tweet'
+    else
+      redirect '/login'
+    end
+  end
+
+  patch '/tweets/:id/edit' do
+    @old_tweet = Tweet.find(params[:id].to_i)
+    if params[:content] != @old_tweet && !params[:content].empty?
+      @old_tweet.content = params[:content]
+    end
+    @old_tweet.save
+  end
+
+  post '/tweets' do
+    @user = User.find(session[:id])
+    if !params[:content].empty?
+      @user.tweets << Tweet.create(content: params[:content])
```

Comment:
Another way to do this is to make use of the `has_many` relationship on User:
```
@tweet = current_user.tweets.create(content: params[:content])
```
Both approaches are good, but in some cases you may find this option useful.